homeassistant/components/yolink/config_flow.py

```diff
@@ -0,0 +1,34 @@
+"""Config flow for yolink."""
+import logging
+
+from homeassistant.helpers import config_entry_oauth2_flow
+
+from .const import DOMAIN
+
+
+class OAuth2FlowHandler(
+    config_entry_oauth2_flow.AbstractOAuth2FlowHandler, domain=DOMAIN
+):
+    """Config flow to handle yolink OAuth2 authentication."""
+
+    DOMAIN = DOMAIN
+
+    @property
+    def logger(self) -> logging.Logger:
+        """Return logger."""
+        return logging.getLogger(__name__)
+
+    @property
+    def extra_authorize_data(self) -> dict:
+        """Extra data that needs to be appended to the authorize url."""
+        scopes = ["create"]
+        return {"scope": " ".join(scopes)}
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow start."""
+        await self.async_set_unique_id(DOMAIN)
+
+        if self._async_current_entries():
+            return self.async_abort(reason="single_instance_allowed")
```

Comment:
Should it only be possible to have a single YoLink account configured? What if they want to add multiple accounts (their account of another location perhaps?)

Reply:
not support currently